frontend/src/component/common/ConstraintsList/ConstraintsList.tsx

```diff
@@ -10,17 +10,21 @@ const StyledList = styled('ul')(({ theme }) => ({
     gap: theme.spacing(1),
 }));
 
-const StyledListItem = styled('li')(({ theme }) => ({
+export const ConstraintListItem = styled('div')(({ theme }) => ({
     position: 'relative',
     border: `1px solid ${theme.palette.divider}`,
     borderRadius: theme.shape.borderRadiusMedium,
     background: theme.palette.background.default,
-    padding: theme.spacing(2, 3),
+    padding: theme.spacing(1.5, 3),
     display: 'flex',
     flexFlow: 'column',
-    gap: theme.spacing(2),
+    gap: theme.spacing(1),
 }));
```

Comment:
Decorating list items with container block (border, padding, background) automatically was a bad idea. It's easier to be more explicit where and how it is used.